02_plotting-mutation-counts.Rmd

```diff
@@ -0,0 +1,265 @@
+
+**A note on formatting:**
+We like to keep our documents formatted with the philosophy of one sentence per line.
+This interacts well with git and GitHub, which record edits based on the line that changed.
+So keeping each sentence on a separate line makes it easier to see what has changed when editing a large block of text; 
+if we just put spaces between sentences within a paragraph, they would appear to git as a single line, and a change anywhere would look like a change to the whole paragraph.
+To separate paragraphs in markdown, we have to use a blank line, so separating sentences onto their own lines won't affect the final output.
+
+This notebook is an example of an analysis that might be performed using the output of the `01_count-gene-mutations.R` script (though with a lot of meta-commentary about the steps and choices made along the way).
+We are assuming that counting was performed for the low-grade astrocytoma (LGAT), and medulloblastoma MAF files for this notebook.
+If you would like, you can make a copy of this notebook, change the input files (and probably some text), and apply the same analysis to different data sets!
+
+## Setup
+
+### Load libraries
+
+```{r libraries}
+library(dplyr)
+library(ggplot2)
+library(magrittr)
+```
+
+### Specify files and paths
+
+We like to specify all our input and output paths toward the top of the notebook, making them easy to find and update as needed.
+During development, we might not know exactly what files are needed at the start, so this may not get *written* first, but having a consistent and prominent place where all paths are defined makes it easy to quickly determine what files are needed and where they will be found.
+
+Since this is an `Rmarkdown` file, all of the file paths will be relative to the location of this file.
+This means that if the file moves, the paths would move, which can be annoying!
+A nice solution we can use is the [`here` package](https://here.r-lib.org), which will allow us to specify paths relative to the project root directory.
+The project root is found by looking for an R project file with the `.Rproj` extension in a parent directory from the file (or a `.git` directory if there is no project file).
+
+We will also take advantage of the fact that the file.path() function (and `here::here()`), allows us to specify paths as a series of comma-separated partial paths, i.e., individual directories and files.
+If we knew everyone were using a UNIX machine, we could use a path like `analysis/gene_counts`, but this is a bit less portable to other systems.
+(Thankfully, paths like this *do* work on Windows now!) Using the separated list of elements also makes it easier to join partial paths together.
+We won't have to worry about whether a partial path ends with a separator slash or not; `file.path()` will add it as needed!
+
+```{r inputs}
+# Set the directory where the counts files are found
+counts_dir <- here::here("analysis", "gene_counts")
```

Comment:
Oh I never actually realized it could be used like this! I have always done, `file.path(here::here(), "analysis", "gene_counts")`. So cool!